zerver/lib/hotspots.py

```diff
@@ -94,50 +32,25 @@ def to_dict(self) -> Dict[str, str]:
     ),
 ]
 
-# We would most likely implement new hotspots in the future that aren't
-# a part of the initial tutorial. To that end, classifying them into
-# categories which are aggregated in ALL_HOTSPOTS, seems like a good start.
-ALL_HOTSPOTS = [*INTRO_HOTSPOTS, *NON_INTRO_HOTSPOTS]
-ALL_ONBOARDING_STEPS: List[Union[Hotspot, OneTimeNotice]] = [*ALL_HOTSPOTS, *ONE_TIME_NOTICES]
+# We may introduce onboarding step of types other than 'one time notice'
+# in future. Earlier, we had 'hotspot' and 'one time notice' as the two
+# types. We can simply do:
+# ALL_ONBOARDING_STEPS: List[Union[OneTimeNotice, OtherType]]
+# to avoid API changes when new type is introduced in the future.
+ALL_ONBOARDING_STEPS: List[OneTimeNotice] = ONE_TIME_NOTICES
 
 
 def get_next_onboarding_steps(user: UserProfile) -> List[Dict[str, Any]]:
-    # For manual testing, it can be convenient to set
-    # ALWAYS_SEND_ALL_HOTSPOTS=True in `zproject/dev_settings.py` to
-    # make it easy to click on all of the hotspots.
-    #
-    # Since this is just for development purposes, it's convenient for us to send
-    # all the hotspots rather than any specific category.
-    if settings.ALWAYS_SEND_ALL_HOTSPOTS:
-        return [hotspot.to_dict() for hotspot in ALL_HOTSPOTS]
-
-    # If a Zulip server has disabled the tutorial, never send hotspots.
-    if not settings.TUTORIAL_ENABLED:
-        return []
```

Comment:
Hmm, this `TUTORIAL_ENABLED` check might still be worth having -- it's intended for more API-centric use cases where one might not want to bother with onboarding content.

Reply:
This was restored by Tim before merging.